modules/run/v9/artifacts/opt/jboss/container/java/run/run-java.sh

```diff
@@ -0,0 +1,228 @@
+    java_opts=$(${JBOSS_CONTAINER_JAVA_JVM_MODULE}/java-default-options)
+  fi
+  if [ -f "${JBOSS_CONTAINER_JAVA_JVM_MODULE}/debug-options" ]; then
+    debug_opts=$(${JBOSS_CONTAINER_JAVA_JVM_MODULE}/debug-options)
+  fi
+  if [ -f "${JBOSS_CONTAINER_JAVA_PROXY_MODULE}/proxy-options" ]; then
+    source "${JBOSS_CONTAINER_JAVA_PROXY_MODULE}/proxy-options"
+    proxy_opts="$(proxy_options)"
+  fi
+  # Normalize spaces with awk (i.e. trim and elimate double spaces)
+  echo "${JAVA_OPTS} $(run_java_options) ${debug_opts} ${proxy_opts} ${java_opts} ${JAVA_OPTS_APPEND}" | awk '$1=$1'
+}
+
+# Read in a classpath either from a file with a single line, colon separated
+# or given line-by-line in separate lines
+# Arg 1: path to claspath (must exist), optional arg2: application jar, which is stripped from the classpath in
+# multi line arrangements
+format_classpath() {
+  local cp_file="$1"
+  local app_jar="$2"
+
+  local wc_out=`wc -l $1 2>&1`
+  if [ $? -ne 0 ]; then
+    log_error "Cannot read lines in ${cp_file}: $wc_out"
+    exit 1
+  fi
+
+  local nr_lines=`echo $wc_out | awk '{ print $1 }'`
+  if [ ${nr_lines} -gt 1 ]; then
+    local sep=""
+    local classpath=""
+    while read file; do
+      local full_path="${JAVA_LIB_DIR}/${file}"
+      # Don't include app jar if include in list
+      if [ x"${app_jar}" != x"${full_path}" ]; then
+        classpath="${classpath}${sep}${full_path}"
+      fi
+      sep=":"
+    done < "${cp_file}"
+    echo "${classpath}"
+  else
+    # Supposed to be a single line, colon separated classpath file
+    cat "${cp_file}"
+  fi
+}
+
+# Fetch classpath from env or from a local "run-classpath" file
+get_classpath() {
+  local cp_path="."
+  if [ "x${JAVA_LIB_DIR}" != "x${JAVA_APP_DIR}" ]; then
+    cp_path="${cp_path}:${JAVA_LIB_DIR}"
+  fi
+  if [ -z "${JAVA_CLASSPATH}" ] && [ "x${JAVA_MAIN_CLASS}" != x ]; then
+    if [ "x${JAVA_APP_JAR}" != x ]; then
+      cp_path="${cp_path}:${JAVA_APP_JAR}"
+    fi
+    if [ -f "${JAVA_LIB_DIR}/classpath" ]; then
+      # Classpath is pre-created and stored in a 'run-classpath' file
+      cp_path="${cp_path}:`format_classpath ${JAVA_LIB_DIR}/classpath ${JAVA_APP_JAR}`"
+    else
+      # No order implied
+      cp_path="${cp_path}:${JAVA_APP_DIR}/*"
+    fi
+  elif [ "x${JAVA_CLASSPATH}" != x ]; then
+    # Given from the outside
+    cp_path="${JAVA_CLASSPATH}"
+  fi
+  echo "${cp_path}"
+}
+
+# Set process name if possible
+get_exec_args() {
+  if [ "x${JAVA_APP_NAME}" != x ]; then
+    echo "-a '${JAVA_APP_NAME}'"
+  fi
+}
+
+# Start JVM
+startup() {
+  # Initialize environment
+  load_env
+
+  local args
+  cd ${JAVA_APP_DIR}
+  if [ "x${JAVA_MAIN_CLASS}" != x ] ; then
+     args="${JAVA_MAIN_CLASS}"
+  else
+     args="-jar ${JAVA_APP_JAR}"
+  fi
+  log_info "exec $(get_exec_args) java $(get_java_options) -cp \"$(get_classpath)\" ${args} $*"
+  exec $(get_exec_args) java $(get_java_options) -cp "$(get_classpath)" ${args} $*
```

Comment:
Consider an option which execs `java SOME_OPTIONS -cp  ...` as an alternative to `java $(get_java_options)...` Where `SOME_OPTIONS` is taken verbatim from user settings.